### PR TITLE
core: do not mark live items as editied until it's no longer live

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

**Review by jr-simplex — APPROVED** [review]

File: src/Simplex/Chat/Store.hs [hunk 1]
```diff
@@ -3965,21 +3967,23 @@ getDirectChatItemIdByText db userId contactId msgDir quotedMsg =
       (userId, contactId, msgDir, quotedMsg <> "%")
 
 updateGroupChatItem :: forall d. MsgDirectionI d => DB.Connection -> User -> Int64 -> ChatItemId -> CIContent d -> Bool -> MessageId -> ExceptT StoreError IO (ChatItem 'CTGroup d)
-updateGroupChatItem db user@User {userId} groupId itemId newContent itemLive msgId = do
-  ci <- liftEither . correctDir =<< getGroupChatItem db user groupId itemId
+updateGroupChatItem db user@User {userId} groupId itemId newContent live msgId = do
+  ci@ChatItem {meta = CIMeta {itemEdited, itemLive}} <- liftEither . correctDir =<< getGroupChatItem db user groupId itemId
   currentTs <- liftIO getCurrentTime
   let newText = ciContentToText newContent
+      edited' = itemEdited || not live
```
Comment (epoberezkin) [inline src/Simplex/Chat/Store.hs:3974]:
```suggestion
      edited' = itemEdited || not itemLive
```

Comment (epoberezkin) [reply src/Simplex/Chat/Store.hs:3974]: fix

File: src/Simplex/Chat/Store.hs [hunk 2]
```diff
@@ -3965,21 +3967,23 @@ getDirectChatItemIdByText db userId contactId msgDir quotedMsg =
       (userId, contactId, msgDir, quotedMsg <> "%")
 
 updateGroupChatItem :: forall d. MsgDirectionI d => DB.Connection -> User -> Int64 -> ChatItemId -> CIContent d -> Bool -> MessageId -> ExceptT StoreError IO (ChatItem 'CTGroup d)
-updateGroupChatItem db user@User {userId} groupId itemId newContent itemLive msgId = do
-  ci <- liftEither . correctDir =<< getGroupChatItem db user groupId itemId
+updateGroupChatItem db user@User {userId} groupId itemId newContent live msgId = do
+  ci@ChatItem {meta = CIMeta {itemEdited, itemLive}} <- liftEither . correctDir =<< getGroupChatItem db user groupId itemId
   currentTs <- liftIO getCurrentTime
   let newText = ciContentToText newContent
+      edited' = itemEdited || not itemLive
```
Comment (epoberezkin) [inline src/Simplex/Chat/Store.hs:3974]:
```suggestion
      edited' = itemEdited || not live
```

File: src/Simplex/Chat/Store.hs [hunk 3]
```diff
@@ -3837,26 +3837,28 @@ updateDirectChatItemStatus db userId contactId itemId itemStatus = do
...
   forM_ msgId_ $ \msgId -> liftIO $ insertChatItemMessage_ db itemId msgId currentTs
   pure ci
 
 updateDirectChatItem_ :: forall d. (MsgDirectionI d) => DB.Connection -> UserId -> Int64 -> ChatItemId -> CIContent d -> Bool -> UTCTime -> ExceptT StoreError IO (ChatItem 'CTDirect d)
-updateDirectChatItem_ db userId contactId itemId newContent itemLive currentTs = do
-  ci <- liftEither . correctDir =<< getDirectChatItem db userId contactId itemId
+updateDirectChatItem_ db userId contactId itemId newContent live currentTs = do
+  ci@ChatItem {meta = CIMeta {itemEdited, itemLive}} <- liftEither . correctDir =<< getDirectChatItem db userId contactId itemId
   let newText = ciContentToText newContent
+      edited' = itemEdited || not itemLive
```
Comment (epoberezkin) [inline src/Simplex/Chat/Store.hs:3850]:
```suggestion
      edited' = itemEdited || not live
```

**Pull request merged**